applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_cr_beam_element_3D2N.cpp

```diff
@@ -0,0 +1,287 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:     BSD License
+//           license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//
+
+
+#include "adjoint_finite_difference_cr_beam_element_3D2N.h"
+#include "structural_mechanics_application_variables.h"
+#include "custom_response_functions/response_utilities/response_data.h"
+#include "includes/checks.h"
+
+
+namespace Kratos
+{
+
+AdjointFiniteDifferenceCrBeamElement::AdjointFiniteDifferenceCrBeamElement(Element::Pointer pPrimalElement)
+    : AdjointFiniteDifferencingBaseElement(pPrimalElement)
+{
+}
+
+AdjointFiniteDifferenceCrBeamElement::~AdjointFiniteDifferenceCrBeamElement()
+{
+}
+
+Element::Pointer AdjointFiniteDifferenceCrBeamElement::Create(Element::Pointer pPrimalElement) const
+{
+    return Kratos::make_shared<AdjointFiniteDifferenceCrBeamElement>(pPrimalElement);
+}
+
+void AdjointFiniteDifferenceCrBeamElement::EquationIdVector(EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo) {
+
+    const int number_of_nodes = this->GetGeometry().PointsNumber();
+    const int dimension = this->GetGeometry().WorkingSpaceDimension();
+    const unsigned int local_size = number_of_nodes * dimension * 2;
+
+    if (rResult.size() != local_size) rResult.resize(local_size);
+
+    GeometryType & geom = this->GetGeometry();
+
+    for (int i = 0; i < number_of_nodes; ++i)
+    {
+        int index = i * number_of_nodes * dimension;
```

Comment:
Shouldn't this be `index = i * dimension * 2`? Even then it could index out of range if dimension=2 since you assume z component below. Doesn't the base class already do this for you?

Reply:
You are right this method can be removed, as it is covered by the base class.

Reply:
Done